fe/fe-core/src/main/java/com/starrocks/connector/iceberg/rest/IcebergRESTCatalog.java

```diff
@@ -168,31 +200,59 @@ public void dropDB(String dbName) throws MetaNotFoundException {
         if (Strings.isNullOrEmpty(dbLocation)) {
             throw new MetaNotFoundException("Database location is empty");
         }
-
-        delegate.dropNamespace(convertDbNameToNamespace(dbName));
+        try {
+            delegate.dropNamespace(convertDbNameToNamespace(dbName));
+        } catch (RESTException re) {
+            LOG.error("Failed to drop database using REST Catalog, for dbName {}", dbName, re);
+            throw new StarRocksConnectorException("Failed to drop database using REST Catalog",
+                    new RuntimeException("Failed to drop database using REST Catalog, exception: " + re.getMessage(), re));
+        }
     }
 
     @Override
     public Database getDB(String dbName) {
-        Map<String, String> dbMeta = delegate.loadNamespaceMetadata(convertDbNameToNamespace(dbName));
-        return new Database(CONNECTOR_ID_GENERATOR.getNextId().asInt(), dbName, dbMeta.get(LOCATION_PROPERTY));
+        try {
+            Map<String, String> dbMeta = delegate.loadNamespaceMetadata(convertDbNameToNamespace(dbName));
+            return new Database(CONNECTOR_ID_GENERATOR.getNextId().asInt(), dbName, dbMeta.get(LOCATION_PROPERTY));
+        } catch (RESTException re) {
+            LOG.error("Failed to get database using REST Catalog, for dbName {}", dbName, re);
+            throw new StarRocksConnectorException("Failed to get database using REST Catalog",
+                    new RuntimeException("Failed to get database using REST Catalog exception:" + re.getMessage(), re));
+        }
     }
 
     @Override
     public List<String> listTables(String dbName) {
         Namespace ns = convertDbNameToNamespace(dbName);
-        List<TableIdentifier> tableIdentifiers = new ArrayList<>(delegate.listTables(ns));
+        List<TableIdentifier> tableIdentifiers = null;
+        try {
+            tableIdentifiers = delegate.listTables(ns);
+        } catch (RESTException re) {
+            LOG.error("Failed to list tables using REST Catalog, for dbName {}", dbName, re);
+            // creating a new RuntimeException with the custom message, as in the upstream code, it picks the `exception.getCause().getMessage()`
```

Comment:
Could we throw exception like this :
```
throw new StarRocksConnectorException("Failed to list tables using REST Catalog, exception:" 
+ re.getMessage(), re);
```

Reply:
@Youngwb problem is when it is printed on the mysql cli (JDBC client) it print the `re.getMessage()` and not the `"Failed to list tables using REST Catalog, exception:" ` of the `StarRocksConnectorException` therefore have to add the custom message in the parent exception.